### ISSUE TITLE
NotImplementedError: Unpack for type ((67108864,)) not implemented

### ISSUE THREAD

**Issue by supermoney007** [dsuch/pymqi#225]
Hello guys, I'm trying to connect to remote queue manager and check queue detph/check channel status, IBM WebSphere MQ has been installed on my windows 7 machine,  may I ask how to deal with the errors below? Thanks in advance.

Source code:

```
import os
import pymqi
import logging

logging.basicConfig(level=logging.INFO)

queue_manager = "SAIC_ESB_MB02"
channel = "SVRCONN"
host = "10.90.11.28"
port = "14500"
conn_info = "%s(%s)" % (host, port)

prefix = '*'
queue_type = pymqi.CMQC.MQQT_LOCAL

args = {pymqi.CMQC.MQCA_Q_NAME: prefix}

qmgr = pymqi.connect(queue_manager, channel, conn_info)
pcf = pymqi.PCFExecute(qmgr)

response = pcf.MQCMD_INQUIRE_Q(args)
qmgr.disconnect()
```

Error message:

```
Traceback (most recent call last):
  File "D:\Tool\test.py", line 21, in <module>
    response = pcf.MQCMD_INQUIRE_Q(args)
  File "C:\Program Files\Python35\lib\site-packages\pymqi-1.12.0-py3.5-win-amd64.egg\pymqi\__init__.py", line 2770, in __call__
    res, mqcfh_response = self.__pcf.unpack(message)
  File "C:\Program Files\Python35\lib\site-packages\pymqi-1.12.0-py3.5-win-amd64.egg\pymqi\__init__.py", line 2991, in unpack
    raise NotImplementedError('Unpack for type ({}) not implemented'.format(pcf_type))
NotImplementedError: Unpack for type ((67108864,)) not implemented
```

**Comment by SeyfSV** [comment]
Hello @supermoney007 ,

It's strange issue, because MQ does not have type with 67108864 value.
I checked v1.12.0 with this test (you can add it to test_pcf.py file to TestPCF lcass and run), I have no issue:
```
    def test_gh_225(self):
        attrs = {
            pymqi.CMQC.MQCA_Q_NAME: '*'
            }

        results = self.pcf.MQCMD_INQUIRE_Q(attrs)

        print(results)
```

Can you provide PCF message or mq trace?
What version of MQ server do you use?

**Comment by SeyfSV** [comment]
Closed due to inactivity

**Comment by agamjajoo** [comment]
Hi @SeyfSV,

I am also seeing this error but only after upgrade to Python3. Same script with Python2.7 is working fine. 
Could it be something to do with Python3? 

Apologies if I am not supposed to comment on closed issues, please let me know if I should open a new issue for this?

Depending upon your recommendation to above i'll either post the details of MQ client version here or in new issue.

Can you provide PCF message or mq trace?
What version of MQ server do you use?

**Comment by SeyfSV** [comment]
@agamjajoo , thank you for info
Yes, behaviour is different between different Python versions, even inside same major version:

> Additionally, CPython 3.6+ uses a new implementation of dictionaries, which makes them appear sorted by insertion order (though you can only rely on this behaviour in Python 3.7+).

I need additional info for reproducing issue, without that I can't help :(.

We are automatically test pymqi with different python versions, but something can go wrong:
```
      matrix:
        environment: ['ubuntu-latest']
        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
        mq-client-version: [9.1.5.0] 
```

**Comment by bhagtrajaram** [comment]
Hi,

Did you guys have a solution for the above? 

/home/gonzales/PycharmProjects/mq2/venv/bin/python /home/gonzales/PycharmProjects/mq/main.py
Traceback (most recent call last):
  File "/home/gonzales/PycharmProjects/mq/main.py", line 19, in <module>
    response = pcf.MQCMD_INQUIRE_Q(args)
  File "/home/gonzales/PycharmProjects/mq2/venv/lib/python2.7/site-packages/pymqi/__init__.py", line 2768, in __call__
    res, mqcfh_response = self.__pcf.unpack(message)
  File "/home/gonzales/PycharmProjects/mq2/venv/lib/python2.7/site-packages/pymqi/__init__.py", line 3007, in unpack
    raise NotImplementedError('Unpack for type ({}) not implemented'.format(pcf_type))
NotImplementedError: Unpack for type ((67108864,)) not implemented

Process finished with exit code 1

**Comment by SeyfSV** [comment]
Hi @bhagtrajaram ,
No, I can't reproduce it in mine and github environments.
Can you provide message ?

**Comment by bhagtrajaram** [comment]
Hi, @SeyfSV,

Unfortunately i don't see anything in de logs on mq.

If i change the following line:
response = pcf.MQCMD_INQUIRE_Q(args)
into:
response = pcf.MQCMD_INQUIRE_Q_MGR(args)

pymqi.MQMIError: MQI Error. Comp: 33554432, Reason 420216832: FAILED: Error code 420216832 not defined

reference:
https://www.ibm.com/support/knowledgecenter/SSFKSJ_8.0.0/com.ibm.mq.ref.adm.doc/q087820_.htm

**Comment by SeyfSV** [comment]
I have some idea... 
What MQCCSID value do you use: qmgr and environment?
Do you have non-latin description on queue?

**Comment by SeyfSV** [comment]
I doesn't reproduce issue, but I think this can help.
@supermoney007 @agamjajoo @bhagtrajaram , please, check version from this branch

**Comment by SeyfSV** [comment]
Is your MQ server running zOS?

**Comment by bhagtrajaram** [comment]
Hi @SeyfSV,

Please find the defails you requested:

- locale

LANG=en_US
LC_COLLATE="en_US"
LC_CTYPE="en_US"
LC_MONETARY="en_US"
LC_NUMERIC="en_US"
LC_TIME="en_US"
LC_MESSAGES="en_US"
LC_ALL=
- locale charmap
ISO8859-1

So the value of the CCSID is ' 819', according to: https://www.ibm.com/support/knowledgecenter/en/SSFKSJ_8.0.0/com.ibm.mq.ref.dev.doc/q104230_.htm.

From my debugger i can see that the client is sending a ccsid of '1208' (default). This is MQ running on AIX.

**Comment by SeyfSV** [comment]
AIX is big-endian operating system (zOS too).
Please, add convert=True

**Comment by SeyfSV** [comment]
I will close this issue due to inactivity
